### PR TITLE
Cmdct 3115 - fix ability to add and delete initiatives

### PR DESCRIPTION
**Pull request by britt-mo**
### Description
After submitting a workplan and having an admin unlock it, a state user couldn't delete initiatives or create or delete new nested initiatives. The issue was due to the fact that every update and delete was sending a full payload with the following 'blacklist` data, which isn't allowed after a workplan has already been submitted/is in revision:
```
    "submitterName",
    "submitterEmailAddress",
    "reportSubmissionDate",
```

### Related ticket(s)

CMDCT-3115

---
### How to test
Create a wp and submit it
login as an admin and unlock it
login as a state user and enter the unlocked workplan
verify that you can delete initiatives and create new ones at the top level and nested level (evaluation plans, funding sources)

### Important updates

---
### Author checklist

- [ ] I have performed a self-review of my code
- [ ] I have added [thorough](https://shorturl.at/aejkF) tests, if necessary
- [ ] I have updated relevant documentation, if necessary
---

_convert to a different template: [test → val](?expand=1&template=test-to-val-deployment.md)_ | _[val → prod](?expand=1&template=val-to-prod-deployment.md)_

### CODE REVIEW

File: services/ui-src/src/components/modals/DeleteEntityModal.tsx [hunk 1]
```diff
@@ -70,18 +70,20 @@ export const DeleteEntityModal = ({
...
     );
 
+    const tempEntities = { ...updatedEntities };
     await updateReport(reportKeys, {
       metadata: {
         lastAlteredBy: full_name,
         status: ReportStatus.IN_PROGRESS,
       },
-      fieldData: { ...updatedEntities },
+      fieldData: { [entityName]: tempEntities[entityName] },
```
Comment (ailZhou) [inline services/ui-src/src/components/modals/DeleteEntityModal.tsx:86]: nit: not sure if its necessary to store the updatedEntities in temp variable when its only for data retrieval and theres no modification after. you can just use it as:

`fieldData: { [entityName]: updatedEntities [entityName] },`

**Review by ailZhou — DISMISSED** [review]
Tested locally and works beautifully and the logic looks sound. I don't see any issues besides one tiny nit but that's not really important. Thank you for fixing this!

**Review by cassandradanger — APPROVED** [review]
looks great in the browser! Thanks for removing those <br>'s in the copy as well :)

File: services/ui-src/src/components/modals/DeleteEntityModal.tsx (hunk 1 above)
Comment (britt-mo) [reply services/ui-src/src/components/modals/DeleteEntityModal.tsx:87]: you were right. updated.

**Review by ailZhou — APPROVED** [review]

**Review by ntsummers1 — APPROVED** [review]

**Pull request merged**